lib/message_builder.dart

```diff
@@ -391,6 +392,17 @@ class MessageBuilder extends PartBuilder {
     }
   }
 
+  bool sign(String privateKey, String domain, String selector) {
+    var msg = buildMimeMessage(),
+        sgn = MailSignature(privateKey, domain, selector).get_signed_headers(msg.bodyRaw ?? msg.parts.map((e) => e.headerRaw + '\r\n' + e.bodyRaw).join('\r\n'), msg.headerRaw);
```

Comment:
Any way to prepare the global bodyRaw for this?
Would help to sign the proper body value

Reply:
Hi Kevin, 

DKIM signing is a useful extension and I look forward to cooperate with you on this. My time in February is quite limited, though - but I'm happy to help where I can.

Please note that I just pushed a rather big change in which the MimeMessage.bodyRaw field is removed and replaced with the mimeData field. This was necessary for supporting 8bit encoded messages with a different encoding than ASCII or UTF8.  

In general I would assume that you need to build a message, then render it and calculate the DKIM based on the message contents. Is that right?

Reply:
Yes the hole body needs to be hashed. My best bet was that "bodyRaw" is already the whole body as it would be sent, but it wasn't. If buildMessage would create such parts "head" and "body" the same way it is sent later - I think that would be enough. And this done somewhere already I guess?